pallets/nfts/src/features/transfer.rs

```diff
@@ -86,6 +86,18 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		// Perform the transfer with custom details using the provided closure.
 		with_details(&collection_details, &mut details)?;
 
+		// Update account balance of the destination account.
+		AccountBalance::<T, I>::mutate(collection, &dest, |balance| {
```

Comment:
Minor: does it not make sense to decrement the owned amount before incrementing the destination account. That is, checking owner has something to transfer before incrementing the balance of the recipient. I appreciate there are checks above that cover this, so not a blocker.

Reply:
I'll reorder the code flow to match with the `Account::remove / insert` below. Resolved.

Reply:
Instead of returning the `ArithmeticError::Underflow`, I return `Error::NoItemsOwned` because I think it makes more sense.